src/golang/lib/airflow/client.go

```diff
@@ -39,15 +39,34 @@ func newClient(ctx context.Context, authConf auth.Config) (*client, error) {
 
 // getDagRuns returns all of the Airflow DAGRuns for the Airflow DAG specified.
 func (c *client) getDagRuns(dagId string) ([]airflow.DAGRun, error) {
-	dagRunsResp, resp, err := c.apiClient.DAGRunApi.GetDagRuns(
-		c.ctx,
-		dagId,
-	).OrderBy("start_date").Execute()
-	if err != nil {
-		return nil, wrapApiError(err, "GetDagRuns", resp)
+	limit := 100 // This is the max number of DAG runs that can be returned in each response.
+	offset := 0
+	var dagRuns []airflow.DAGRun
+
+	// Keep paginating through DAG runs until there are none in response
+	for {
+		dagRunsResp, resp, err := c.apiClient.DAGRunApi.GetDagRuns(
+			c.ctx,
+			dagId,
+		).
+			OrderBy("start_date").
```

Comment:
off topic - this looks really cool, idk how hard it is for us to implement something similar. Though this might be backed by graphQL or ORM when it gets to query management, and we don't want to do either for now